k8s/upgrade/src/preflight_validations.rs

```diff
@@ -168,15 +168,29 @@ pub async fn is_rebuild_in_progress(client: &upgrade_lib::RestClient) -> error::
 /// Struct to deserialize the unsupported version yaml.
 #[derive(Debug, Serialize, Deserialize)]
 struct UnsupportedVersions {
-    unsupported_versions: Vec<String>,
+    unsupported_versions: Vec<Version>,
 }
 
-pub async fn upgrade_path_validation(namespace: &str) -> error::Result<()> {
-    let path = std::env::current_dir()
-        .context(error::GetCurrentDirectory)?
-        .join(UNSUPPORTED_VERSION_FILE);
+impl UnsupportedVersions {
+    fn contains(&self, v: &Version) -> bool {
+        self.unsupported_versions.contains(v)
+    }
+}
+
+impl TryFrom<&[u8]> for UnsupportedVersions {
+    type Error = crate::error::Error;
 
-    let unsupported_versions = get_unsupported_versions(path)?;
+    /// Returns an UnsupportedVersions object.
+    fn try_from(bytes: &[u8]) -> error::Result<Self> {
+        let mut path = PathBuf::new();
+        path.push(UNSUPPORTED_VERSION_FILE);
+        serde_yaml::from_reader(bytes).context(error::YamlParseFromFile { filepath: path })
+    }
+}
+
+pub async fn upgrade_path_validation(namespace: &str) -> error::Result<()> {
+    let unsupported_version_file = &std::include_bytes!("../config/unsupported_versions.yaml")[..];
```

Comment:
nit: This variable is misleading. This is not a std::fs::File. Maybe rename to unsupported_version_buf? As in... it is a bytes buffer.